gramex/pptgen2/__init__.py

```diff
@@ -103,8 +104,33 @@ def pptgen(source: Union[str, pptx.presentation.Presentation],
                 slide_data['slide'] = slide         # Rule can use 'slide' as a variable
                 transition(slide, rule.get('transition', None), data)
                 apply_commands(rule, slide.shapes, slide_data)
+
+    #  requirements: install LibreOffice, ImageMagick
+    #  (https://learn.gramener.com/guide/pptxhandler/#pptx-to-images)
+
+    #  Ensure both LibreOffice, ImageMagick are updated in the environment variables
+    #  C:\Program Files\LibreOffice\program, C:\Program Files\ImageMagick-7.0.11-Q16-HDRI
+    #  Add fonts to LiberOffice if they are not available by default
+
+    #  Get filename, extension, save the file as pptx first
+    pres_name, pdf_name, file_ext = None, None, None
     if target:
-        prs.save(target)
+        file_name, file_ext = os.path.splitext(target)
+        pres_name = f'{file_name}.pptx'
+        prs.save(pres_name)
```

Comment:
This should be `prs.save(target)` -- what if target has a `.PPTX` in caps? Just retain the original.

Instead, just extract the `file_ext` -- convert it to lowercase.